File: create_tables/anl_accuracy.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table anl_accuracy';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table anl_accuracy(
    ANALYSIS_ID VARCHAR(40) PRIMARY KEY NOT NULL,
    ANL_SOURCE VARCHAR(40) NOT NULL,
    ACCURACY_OBS_NO INT,
    ACCURACY_DESC VARCHAR(240),
    ACCURACY_TYPE VARCHAR(40),
    ACTIVE_IND VARCHAR(1),
    CALCULATED_IND VARCHAR(1),
    CALCULATE_METHOD_ID VARCHAR(40),
    CONFIDENCE_TYPE VARCHAR(40),
    EFFECTIVE_DATE TIMESTAMP(0),
    EXPIRY_DATE TIMESTAMP(0),
    MEASURED_VALUE_IND VARCHAR(1),
    PPDM_GUID VARCHAR(38),
    RAW_VALUE_IND VARCHAR(1),
    REFERENCED_COLUMN_NAME VARCHAR(30),
    REFERENCED_PPDM_GUID VARCHAR(38),
    REFERENCED_SYSTEM_ID VARCHAR(40),
    REFERENCED_TABLE_NAME VARCHAR(30),
    REMARK VARCHAR(2000),
    REPEATABILITY_TYPE VARCHAR(40),
    REPORTED_IND VARCHAR(1),
    SOURCE VARCHAR(40),
    STEP_SEQ_NO INT,
    ROW_CHANGED_BY VARCHAR(30),
    ROW_CHANGED_DATE TIMESTAMP(0),
    ROW_CREATED_BY VARCHAR(30),
    ROW_CREATED_DATE TIMESTAMP(0),
    ROW_EFFECTIVE_DATE TIMESTAMP(0),
    ROW_EXPIRY_DATE TIMESTAMP(0),
    ROW_QUALITY VARCHAR(40))""")

File: postgres_39/all_struct_with_default/AnlAccuracy.go
package dto

type Anl_accuracy struct {
	Analysis_id            string  `json:"analysis_id" default:"source"`
	Anl_source             string  `json:"anl_source" default:"source"`
	Accuracy_obs_no        int     `json:"accuracy_obs_no" default:"1"`
	Accuracy_desc          *string `json:"accuracy_desc" default:""`
	Accuracy_type          *string `json:"accuracy_type" default:""`
	Active_ind             *string `json:"active_ind" default:""`
	Calculated_ind         *string `json:"calculated_ind" default:""`
	Calculate_method_id    *string `json:"calculate_method_id" default:""`
	Confidence_type        *string `json:"confidence_type" default:""`
	Effective_date         *string `json:"effective_date" default:""`
	Expiry_date            *string `json:"expiry_date" default:""`
	Measured_value_ind     *string `json:"measured_value_ind" default:""`
	Ppdm_guid              *string `json:"ppdm_guid" default:""`
	Raw_value_ind          *string `json:"raw_value_ind" default:""`
	Referenced_column_name *string `json:"referenced_column_name" default:""`
	Referenced_ppdm_guid   *string `json:"referenced_ppdm_guid" default:""`
	Referenced_system_id   *string `json:"referenced_system_id" default:""`
	Referenced_table_name  *string `json:"referenced_table_name" default:""`
	Remark                 *string `json:"remark" default:""`
	Repeatability_type     *string `json:"repeatability_type" default:""`
	Reported_ind           *string `json:"reported_ind" default:""`
	Source                 *string `json:"source" default:""`
	Step_seq_no            *int    `json:"step_seq_no" default:""`
	Row_changed_by         *string `json:"row_changed_by" default:""`
	Row_changed_date       *string `json:"row_changed_date" default:""`
	Row_created_by         *string `json:"row_created_by" default:""`
	Row_created_date       *string `json:"row_created_date" default:""`
	Row_effective_date     *string `json:"row_effective_date" default:""`
	Row_expiry_date        *string `json:"row_expiry_date" default:""`
	Row_quality            *string `json:"row_quality" default:""`
}
